{
  boolean[] x={true};
  boolean ret=false;
  boolean thrown=false;
  try {
    ret=Array.getBoolean(x,0);
  }
 catch (  Exception e) {
    fail("Exception during get test : " + e.getMessage());
  }
  assertTrue("Get returned incorrect value",ret);
  try {
    ret=Array.getBoolean(new Object(),0);
  }
 catch (  IllegalArgumentException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Passing non-array failed to throw exception");
  }
  thrown=false;
  try {
    ret=Array.getBoolean(x,4);
  }
 catch (  ArrayIndexOutOfBoundsException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Invalid index failed to throw exception");
  }
  thrown=false;
  try {
    ret=Array.getBoolean(null,0);
  }
 catch (  NullPointerException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Null argument failed to throw NPE");
  }
}
